test/integration/targets/aws_glacier_vault/tasks/main.yml

```diff
@@ -0,0 +1,106 @@
+---
+#
+#  Author: Michael De La Rue
+#  based on ec2_key.yml + lambda.py
+
+- block:
+
+    # ============================================================
+    - name: test with no parameters
+      aws_glacier_vault:
+      register: result
+      ignore_errors: true
+
+    - name: assert failure when called with no parameters
+      assert:
+        that:
+           - 'result.failed'
+           - 'result.msg.startswith("missing required arguments: name")'
+
+    # ============================================================
+    - name: test state=present without tags (expect changed=True)
+      aws_glacier_vault:
+        name='{{glacier_vault_name}}'
+        ec2_region='{{ec2_region}}'
```

Comment:
This is old style formatting.

Have a look at ec2_group for the most recent approach, using YAML anchors to remove repetition

Reply:
i don't like it either, i am happy you prefer the yaml style and will change this without problem.